calculator/prompt.h

```diff
@@ -11,6 +11,8 @@
 
 #include <stdio.h>
 
-int prompt(void);
+typedef char *(*computer)(char *);
```

Comment:
This defines a new type which is a function pointer. In this case we are  saying that the `compute` type should be a pointer to a function which receives a `char *` as its first parameter and returns a `char *`. We do this because it will allow us to separate out the computation work from the prompt control flow.